contract_payment_mode/models/contract.py

```diff
@@ -0,0 +1,26 @@
+from odoo import api, fields, models
+
+
+class AccountAnalyticAccount(models.Model):
+    _inherit = 'account.analytic.account'
+
+    payment_mode_id = fields.Many2one(
+        comodel_name='account.payment.mode',
+        string='Payment Mode',
+        domain=[('payment_type', '=', 'inbound')],
+    )
+
+    @api.onchange('partner_id')
+    def on_change_partner_id(self):
+        if self.partner_id.customer_payment_mode_id:
```

Comment:
Remove condition, if contract has a payment_mode and now the new customer selected has not one, the contract payyment_mode must be False